akka-stream/src/main/scala/akka/stream/stage/GraphStage.scala

```diff
@@ -61,6 +61,24 @@ abstract class GraphStage[S <: Shape] extends GraphStageWithMaterializedValue[S,
   def createLogic: GraphStageLogic
 }
 
+/**
+ * A SourceStage represents a reusable graph stream processing stage. A SourceStage consists of a [[akka.stream.Shape]] which describes
+ * its output port.
+ */
+abstract class SourceStage[Out, M](name: String) extends GraphStageWithMaterializedValue[SourceShape[Out], M] {
```

Comment:
ah no, that method should be on a subclass of GraphStageLogic, but nevermind we can do this later.